From 6ea42d4fa6f3f6dd18c37cb7b6e1faea6afefba4 Mon Sep 17 00:00:00 2001
From: "clopez@igalia.com" <clopez@igalia.com>
Date: Tue, 15 Sep 2015 21:50:18 +0000
Subject: [PATCH] [GTK] Build failure with ACCELERATED_2D_CANVAS when cairo-gl has
 built with OpenGLESv2 support only.

When cairo-gl is built with GLX and ACCELERATED_2D_CANVAS if cairo-gl was only
built with OpenGLESv2, cairo-glx is not enabled causing
Source/WebCore/platform/graphics/glx/GLContextGLX.cpp to reference an undeclared
function and cause a compliation error. Adding an extra check resolves this
build failure.

Upstream-Status: Backport [webkit-2.10.0]

Signed-off-by: Pushpal Sidhu <psidhu@gateworks.com>
---
 Source/WebCore/platform/graphics/glx/GLContextGLX.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/graphics/glx/GLContextGLX.cpp b/Source/WebCore/platform/graphics/glx/GLContextGLX.cpp
index 7890d8d..4ed3a43 100644
--- a/Source/WebCore/platform/graphics/glx/GLContextGLX.cpp
+++ b/Source/WebCore/platform/graphics/glx/GLContextGLX.cpp
@@ -266,7 +266,7 @@ cairo_device_t* GLContextGLX::cairoDevice()
     if (m_cairoDevice)
         return m_cairoDevice;
 
-#if ENABLE(ACCELERATED_2D_CANVAS)
+#if ENABLE(ACCELERATED_2D_CANVAS) && CAIRO_HAS_GLX_FUNCTIONS
     m_cairoDevice = cairo_glx_device_create(sharedX11Display(), m_context);
 #endif
 
-- 
2.6.2

